pkg/generate/app/dockerimagelookup.go

```diff
@@ -41,13 +40,18 @@ type DockerClientSearcher struct {
 }
 
 // Search searches all images in local docker server for images that match terms
-func (r DockerClientSearcher) Search(terms ...string) (ComponentMatches, error) {
+func (r DockerClientSearcher) Search(precise bool, terms ...string) (ComponentMatches, []error) {
 	componentMatches := ComponentMatches{}
 	errs := []error{}
 	for _, term := range terms {
+		if term == "__dockerimage_fail" {
+			errs = append(errs, fmt.Errorf("unable to find the specified docker image: %s", term))
+			continue
+		}
+
 		ref, err := imageapi.ParseDockerImageReference(term)
 		if err != nil {
-			return nil, err
+			continue
```

Comment:
But we show you the typo.  Returning an error should actually be an error
that applies to the term - a term that can't be an image isn't really an
error.  "../foo/bar/template.json" isn't an error

On Tue, Feb 2, 2016 at 3:21 PM, Ben Parees <notifications@github.com> wrote:

> In pkg/generate/app/dockerimagelookup.go
> <https://github.com/openshift/origin/pull/6670#discussion_r51628839>:
>
> >  		ref, err := imageapi.ParseDockerImageReference(term)
> >  		if err != nil {
> > -			return nil, err
> > +			continue
>
> as a user i feel like i'd want to be told that, rather than repeatedly
> being told "nothing matched your stupid typo"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6670/files#r51628839>.
>